codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/traits/SyntheticStructureFromConstrainedMemberTrait.kt

```diff
@@ -0,0 +1,14 @@
+package software.amazon.smithy.rust.codegen.server.smithy.traits
+
+import software.amazon.smithy.model.node.Node
+import software.amazon.smithy.model.shapes.ShapeId
+import software.amazon.smithy.model.traits.AnnotationTrait
+
+/**
+ * Trait applied to a refactored shape indicating the structure that contains member of this new structure type
+ */
+class SyntheticStructureFromConstrainedMemberTrait(val sourceMember : ShapeId): AnnotationTrait(SyntheticStructureFromConstrainedMemberTrait.ID, Node.objectNode())  {
```

Comment:
I am not sure calling `model.toBuilder()` and then calling `toBuild()` on it retains the existing JVM shape object. But I will try this out and if the object remains the same, will keep the shape instead of the ID.

Reply:
We keep the actual `Shapes` in the `SyntheticTrait` now.